.github/workflows/build_and_test.yaml

```diff
@@ -53,29 +52,68 @@ jobs:
         name: codecov-envoy-gateway
         verbose: true
 
-    # build
-    - name: Build Multiarch EG Binaries
+  build:
+    runs-on: ubuntu-latest
+    needs: [lint, gen-check, license-check, coverage-test]
+    steps:
+    - uses: actions/checkout@v3
+    - uses: ./tools/github-actions/setup-deps
+
+    - name: Build EG Multiarch Binaries
       run: make build-multiarch PLATFORMS="linux_amd64 linux_arm64"
 
-    # conformance
-    - name: Run Conformance Tests (v1.24.0)
-      env:
-        KIND_NODE_TAG: v1.24.0
-        CONFORMANCE_UNIQUE_PORTS: false
-      run: make conformance
-    
-    - name: Run Conformance Tests (v1.23.6)
-      env:
-        KIND_NODE_TAG: v1.23.6
-        CONFORMANCE_UNIQUE_PORTS: false
-      run: make conformance
+    - name: Upload EG Binaries
+      uses: actions/upload-artifact@v3
+      with:
+        name: envoy-gateway
+        path: bin/
+
+  conformance-test:
+    runs-on: ubuntu-latest
+    needs: [build]
+    strategy:
+      matrix:
+        version: [ v1.22.9, v1.23.6, v1.24.0 ]
+    steps:
+    - uses: actions/checkout@v3
+    - uses: ./tools/github-actions/setup-deps
+
+    - name: Download EG Binaries
+      uses: actions/download-artifact@v3
+      with:
+        name: envoy-gateway
+        path: bin/
     
-    - name: Run Conformance Tests (v1.22.9)
+    - name: Give Privileges To EG Binaries
```

Comment:
should this be moved into the `build` stage ?

Reply:
Nope, the build stage works fine. The problem is that the upload/download action will break binaries privileges, so we can just give them privileges after download them from GitHub action,